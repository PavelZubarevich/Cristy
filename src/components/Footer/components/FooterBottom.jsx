import { useTranslation } from 'react-i18next';
import { FaLongArrowAltRight } from 'react-icons/fa';

function FooterBottom() {
  const { t } = useTranslation();
  return (
    <div className="footer__bottom">
      <div className="container">
        <div className="footer__bottomContent">
          <p className="footer__copyright">
            {t('CopyrightText')}
          </p>
          <form action="#" className="footer__subscribe">
            <input type="text" name="" placeholder={t('SubscribeToNewsText')} />
            <button type="submit">
              <FaLongArrowAltRight />
            </button>
          </form>
        </div>
      </div>
    </div>
  );
}

export default FooterBottom;
