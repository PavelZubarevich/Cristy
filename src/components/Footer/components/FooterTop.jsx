import { FaFacebookF } from 'react-icons/fa';
import { IoLogoInstagram, IoLogoTwitter } from 'react-icons/io';
import { useTranslation } from 'react-i18next';

function FooterTop() {
  const { t } = useTranslation();
  return (
    <div className="footer__top">
      <div className="container">
        <div className="footer__topContent">
          <div className="footer__topColumn">
            <p className="footer__navCategory">{t('CompanyLabel')}</p>
            <ul className="footer__links">
              <li className="footer__link"><a href="/about">{t('AboutLabel')}</a></li>
              <li className="footer__link"><a href="/shops">{t('OurShopsLabel')}</a></li>
              <li className="footer__link"><a href="/news">{t('NewsLabel')}</a></li>
              <li className="footer__link"><a href="/devivery">{t('DeliveryLabel')}</a></li>
            </ul>
          </div>
          <div className="footer__topColumn">
            <p className="footer__navCategory">{t('HelpLabel')}</p>
            <ul className="footer__links">
              <li className="footer__link"><a href="/contacts">{t('ContactUsLabel')}</a></li>
              <li className="footer__link"><a href="/faq">{t('FAQLabel')}</a></li>
              <li className="footer__link"><a href="/help">{t('HelpLabel')}</a></li>
              <li className="footer__link"><a href="/terms">{t('TermsLabel')}</a></li>
            </ul>
          </div>
          <div className="footer__topColumn">
            <p className="footer__navCategory">{t('GetSocialLabel')}</p>
            <ul className="footer__links">
              <li className="footer__link">
                <a href="/about">
                  <FaFacebookF />
                  Facebook
                </a>
              </li>
              <li className="footer__link">
                <a href="/about">
                  <IoLogoTwitter />
                  Twitter
                </a>
              </li>
              <li className="footer__link">
                <a href="/about">
                  <IoLogoInstagram />
                  Instagram
                </a>
              </li>
            </ul>
          </div>
          <div className="footer__topColumn">
            <p className="footer__navCategory">{t('ContactLabel')}</p>
            <ul className="footer__links">
              <li className="footer__link">
                7/4 Fire Briagde Road
                Khulna, 9100, Bangladesh
              </li>
              <li className="footer__link">
                <strong>
                  {t('PhoneLabel')}
                  :
                </strong>
                <a href="tel:123456789">123-456-789</a>
              </li>
              <li className="footer__link">
                <strong>
                  {t('EmailLabel')}
                  :
                </strong>
                <a href="mailto:support@domain.com">support@domain.com</a>
              </li>
            </ul>
          </div>
        </div>
      </div>

    </div>
  );
}

export default FooterTop;
