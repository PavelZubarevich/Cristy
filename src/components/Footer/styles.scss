@import '../../styles/vars.scss';

.footer {
  border-top: 1px solid $secondaryText;

  &__top {
    background-color: #fff;
  }

  &__topContent {
    display: flex;
    justify-content: space-between;
    gap: 25px;
    padding: 75px 0;
  }

  &__topColumn {
    flex: 0 1 25%;
  }

  &__navCategory {
    font-family: 'Lora';
    font-weight: 700;
    font-size: 20px;
    color: $primaryDarkText;
    margin-bottom: 15px;
  }

  &__link {
    font-family: 'Raleway';
    font-weight: 400;
    font-size: 14px;
    color: $primaryText;
    margin-bottom: 10px;
    display: flex;
    gap: 7px;

    &>a {
      font-family: inherit;
      font-weight: inherit;
      color: inherit;
      transition: 0.3s;
      display: flex;
      align-items: center;

      &>svg {
        font-size: 21px;
        margin-right: 10px;
      }

      &:hover {
        color: #000;
        transform: scale(1.05);
      }
    }
  }

  &__bottom {
    background-color: $secondaryDarkDark;
  }

  &__bottomContent {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 80px 0;
  }

  &__copyright {
    font-family: 'Lato';
    font-size: 15px;
    color: $secondaryText;
  }

  &__subscribe {
    position: relative;
    border-bottom: 1px solid $secondaryText;
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 270px;

    &>input {
      flex: 1;
      background-color: transparent;
      border: none;
      padding: 10px 5px 10px 0;
      position: relative;
      color: $secondaryText;

      &::placeholder {
        color: $secondaryText;
        opacity: 1;
      }
    }

    &>button {
      color: $primary;
      font-size: 18px;
      background-color: transparent;
      border: none;
      cursor: pointer;
      display: flex;
      align-items: center;
    }
  }
}