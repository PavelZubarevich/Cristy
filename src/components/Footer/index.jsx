import './styles.scss';
import FooterBottom from './components/FooterBottom';
import FooterTop from './components/FooterTop';

function Footer() {
  return (
    <footer>
      <FooterTop />
      <FooterBottom />
    </footer>
  );
}

export default Footer;
