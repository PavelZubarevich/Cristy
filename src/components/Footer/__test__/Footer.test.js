import { render } from '@testing-library/react';
import Footer from '..';
import FooterTop from '../components/FooterTop';
import FooterBottom from '../components/FooterBottom';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str) => str,
    i18n: {
      changeLanguage: () => new Promise(() => { }),
      options: {
        supportedLngs: ['ru', 'en'],
      },
    },
  }),
}));

describe('Footer', () => {
  it('Footer rendered', () => {
    render(<Footer />);
    expect(document.body).toMatchSnapshot();
  });
});

describe('Footer top', () => {
  it('Footer rendered', () => {
    render(<FooterTop />);
    expect(document.body).toMatchSnapshot();
  });
});

describe('Footer bottom', () => {
  it('Footer rendered', () => {
    render(<FooterBottom />);
    expect(document.body).toMatchSnapshot();
  });
});
