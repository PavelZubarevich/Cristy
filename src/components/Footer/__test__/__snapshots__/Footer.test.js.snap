// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer rendered 1`] = `
<body>
  <div>
    <footer
      class="footer"
    >
      <div
        class="footer__top"
      >
        <div
          class="container"
        >
          <div
            class="footer__topContent"
          >
            <div
              class="footer__topColumn"
            >
              <p
                class="footer__navCategory"
              >
                CompanyLabel
              </p>
              <ul
                class="footer__links"
              >
                <li
                  class="footer__link"
                >
                  <a
                    href="/about"
                  >
                    AboutLabel
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <a
                    href="/shops"
                  >
                    OurShopsLabel
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <a
                    href="/news"
                  >
                    NewsLabel
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <a
                    href="/devivery"
                  >
                    DeliveryLabel
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="footer__topColumn"
            >
              <p
                class="footer__navCategory"
              >
                HelpLabel
              </p>
              <ul
                class="footer__links"
              >
                <li
                  class="footer__link"
                >
                  <a
                    href="/contacts"
                  >
                    ContactUsLabel
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <a
                    href="/faq"
                  >
                    FAQLabel
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <a
                    href="/help"
                  >
                    HelpLabel
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <a
                    href="/terms"
                  >
                    TermsLabel
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="footer__topColumn"
            >
              <p
                class="footer__navCategory"
              >
                GetSocialLabel
              </p>
              <ul
                class="footer__links"
              >
                <li
                  class="footer__link"
                >
                  <a
                    href="/about"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 320 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                      />
                    </svg>
                    Facebook
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <a
                    href="/about"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M492 109.5c-17.4 7.7-36 12.9-55.6 15.3 20-12 35.4-31 42.6-53.6-18.7 11.1-39.4 19.2-61.5 23.5C399.8 75.8 374.6 64 346.8 64c-53.5 0-96.8 43.4-96.8 96.9 0 7.6.8 15 2.5 22.1-80.5-4-151.9-42.6-199.6-101.3-8.3 14.3-13.1 31-13.1 48.7 0 33.6 17.2 63.3 43.2 80.7-16-.4-31-4.8-44-12.1v1.2c0 47 33.4 86.1 77.7 95-8.1 2.2-16.7 3.4-25.5 3.4-6.2 0-12.3-.6-18.2-1.8 12.3 38.5 48.1 66.5 90.5 67.3-33.1 26-74.9 41.5-120.3 41.5-7.8 0-15.5-.5-23.1-1.4C62.8 432 113.7 448 168.3 448 346.6 448 444 300.3 444 172.2c0-4.2-.1-8.4-.3-12.5C462.6 146 479 129 492 109.5z"
                      />
                    </svg>
                    Twitter
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <a
                    href="/about"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M336 96c21.2 0 41.3 8.4 56.5 23.5S416 154.8 416 176v160c0 21.2-8.4 41.3-23.5 56.5S357.2 416 336 416H176c-21.2 0-41.3-8.4-56.5-23.5S96 357.2 96 336V176c0-21.2 8.4-41.3 23.5-56.5S154.8 96 176 96h160m0-32H176c-61.6 0-112 50.4-112 112v160c0 61.6 50.4 112 112 112h160c61.6 0 112-50.4 112-112V176c0-61.6-50.4-112-112-112z"
                      />
                      <path
                        d="M360 176c-13.3 0-24-10.7-24-24s10.7-24 24-24c13.2 0 24 10.7 24 24s-10.8 24-24 24zM256 192c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64m0-32c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96z"
                      />
                    </svg>
                    Instagram
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="footer__topColumn"
            >
              <p
                class="footer__navCategory"
              >
                ContactLabel
              </p>
              <ul
                class="footer__links"
              >
                <li
                  class="footer__link"
                >
                  7/4 Fire Briagde Road Khulna, 9100, Bangladesh
                </li>
                <li
                  class="footer__link"
                >
                  <strong>
                    PhoneLabel
                    :
                  </strong>
                  <a
                    href="tel:123456789"
                  >
                    123-456-789
                  </a>
                </li>
                <li
                  class="footer__link"
                >
                  <strong>
                    EmailLabel
                    :
                  </strong>
                  <a
                    href="mailto:support@domain.com"
                  >
                    support@domain.com
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer__bottom"
      >
        <div
          class="container"
        >
          <div
            class="footer__bottomContent"
          >
            <p
              class="footer__copyright"
            >
              CopyrightText
            </p>
            <form
              action="#"
              class="footer__subscribe"
            >
              <input
                name=""
                placeholder="SubscribeToNewsText"
                type="text"
              />
              <button
                type="submit"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
                  />
                </svg>
              </button>
            </form>
          </div>
        </div>
      </div>
    </footer>
  </div>
</body>
`;

exports[`Footer bottom Footer rendered 1`] = `
<body>
  <div>
    <div
      class="footer__bottom"
    >
      <div
        class="container"
      >
        <div
          class="footer__bottomContent"
        >
          <p
            class="footer__copyright"
          >
            CopyrightText
          </p>
          <form
            action="#"
            class="footer__subscribe"
          >
            <input
              name=""
              placeholder="SubscribeToNewsText"
              type="text"
            />
            <button
              type="submit"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
                />
              </svg>
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Footer top Footer rendered 1`] = `
<body>
  <div>
    <div
      class="footer__top"
    >
      <div
        class="container"
      >
        <div
          class="footer__topContent"
        >
          <div
            class="footer__topColumn"
          >
            <p
              class="footer__navCategory"
            >
              CompanyLabel
            </p>
            <ul
              class="footer__links"
            >
              <li
                class="footer__link"
              >
                <a
                  href="/about"
                >
                  AboutLabel
                </a>
              </li>
              <li
                class="footer__link"
              >
                <a
                  href="/shops"
                >
                  OurShopsLabel
                </a>
              </li>
              <li
                class="footer__link"
              >
                <a
                  href="/news"
                >
                  NewsLabel
                </a>
              </li>
              <li
                class="footer__link"
              >
                <a
                  href="/devivery"
                >
                  DeliveryLabel
                </a>
              </li>
            </ul>
          </div>
          <div
            class="footer__topColumn"
          >
            <p
              class="footer__navCategory"
            >
              HelpLabel
            </p>
            <ul
              class="footer__links"
            >
              <li
                class="footer__link"
              >
                <a
                  href="/contacts"
                >
                  ContactUsLabel
                </a>
              </li>
              <li
                class="footer__link"
              >
                <a
                  href="/faq"
                >
                  FAQLabel
                </a>
              </li>
              <li
                class="footer__link"
              >
                <a
                  href="/help"
                >
                  HelpLabel
                </a>
              </li>
              <li
                class="footer__link"
              >
                <a
                  href="/terms"
                >
                  TermsLabel
                </a>
              </li>
            </ul>
          </div>
          <div
            class="footer__topColumn"
          >
            <p
              class="footer__navCategory"
            >
              GetSocialLabel
            </p>
            <ul
              class="footer__links"
            >
              <li
                class="footer__link"
              >
                <a
                  href="/about"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 320 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                    />
                  </svg>
                  Facebook
                </a>
              </li>
              <li
                class="footer__link"
              >
                <a
                  href="/about"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M492 109.5c-17.4 7.7-36 12.9-55.6 15.3 20-12 35.4-31 42.6-53.6-18.7 11.1-39.4 19.2-61.5 23.5C399.8 75.8 374.6 64 346.8 64c-53.5 0-96.8 43.4-96.8 96.9 0 7.6.8 15 2.5 22.1-80.5-4-151.9-42.6-199.6-101.3-8.3 14.3-13.1 31-13.1 48.7 0 33.6 17.2 63.3 43.2 80.7-16-.4-31-4.8-44-12.1v1.2c0 47 33.4 86.1 77.7 95-8.1 2.2-16.7 3.4-25.5 3.4-6.2 0-12.3-.6-18.2-1.8 12.3 38.5 48.1 66.5 90.5 67.3-33.1 26-74.9 41.5-120.3 41.5-7.8 0-15.5-.5-23.1-1.4C62.8 432 113.7 448 168.3 448 346.6 448 444 300.3 444 172.2c0-4.2-.1-8.4-.3-12.5C462.6 146 479 129 492 109.5z"
                    />
                  </svg>
                  Twitter
                </a>
              </li>
              <li
                class="footer__link"
              >
                <a
                  href="/about"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M336 96c21.2 0 41.3 8.4 56.5 23.5S416 154.8 416 176v160c0 21.2-8.4 41.3-23.5 56.5S357.2 416 336 416H176c-21.2 0-41.3-8.4-56.5-23.5S96 357.2 96 336V176c0-21.2 8.4-41.3 23.5-56.5S154.8 96 176 96h160m0-32H176c-61.6 0-112 50.4-112 112v160c0 61.6 50.4 112 112 112h160c61.6 0 112-50.4 112-112V176c0-61.6-50.4-112-112-112z"
                    />
                    <path
                      d="M360 176c-13.3 0-24-10.7-24-24s10.7-24 24-24c13.2 0 24 10.7 24 24s-10.8 24-24 24zM256 192c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64m0-32c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96z"
                    />
                  </svg>
                  Instagram
                </a>
              </li>
            </ul>
          </div>
          <div
            class="footer__topColumn"
          >
            <p
              class="footer__navCategory"
            >
              ContactLabel
            </p>
            <ul
              class="footer__links"
            >
              <li
                class="footer__link"
              >
                7/4 Fire Briagde Road Khulna, 9100, Bangladesh
              </li>
              <li
                class="footer__link"
              >
                <strong>
                  PhoneLabel
                  :
                </strong>
                <a
                  href="tel:123456789"
                >
                  123-456-789
                </a>
              </li>
              <li
                class="footer__link"
              >
                <strong>
                  EmailLabel
                  :
                </strong>
                <a
                  href="mailto:support@domain.com"
                >
                  support@domain.com
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
